air/src/stack/op_flags/mod.rs

```diff
@@ -62,12 +79,28 @@ impl<E: FieldElement> OpFlags<E> {
 
     /// Returns a new instance of [OpFlags] instantiated with all stack operation flags.
     pub fn new(frame: &EvaluationFrame<E>) -> Self {
-        let (degree_6_op_flags, degree_4_op_flags) = set_op_flags_degree_4_and_6(frame);
+        let (
+            degree_7_op_flags,
+            degree_6_op_flags,
+            degree_4_op_flags,
+            no_change_flags,
+            left_shift_at_flags,
+            right_shift_at_flags,
+            left_shift_flag,
+            right_shift_flag,
+            control_flow_flag,
+        ) = generate_op_flags(frame);
 
         Self {
-            degree7_op_flags: set_op_flags_degree_7(frame),
+            degree7_op_flags: degree_7_op_flags,
             degree6_op_flags: degree_6_op_flags,
             degree4_op_flags: degree_4_op_flags,
+            no_shift_flags: no_change_flags,
+            left_shift_flags: left_shift_at_flags,
+            right_shift_flags: right_shift_at_flags,
+            left_shift_flag,
+            right_shift_flag,
+            control_flow_flag,
         }
```

Comment:
This code seems a bit redundant: should we just move the content of `generate_op_flags` into the constructor here?

Reply:
But then it will make the constructor messier. Let me know if we should move it in the constructor.

Reply:
I think that's fine - in the end it is the same function just located in a different place in the file.